app/models/participant_profile_state.rb

```diff
@@ -27,4 +27,8 @@ def valid?(_context = nil)
   }
 
   scope :most_recent, -> { order("created_at desc").limit(1) }
+
+  after_create do |profile_state|
```

Comment:
Suggestion: I would strongly advise no to use active record callback, this usually leads to spaghetti code and hide the complexity deep into the database layer.
A more streamlined solution to active record callbacks are to do this inside the transaction were the record is created or updated.